libraries/freertos_plus/standard/https/include/iot_https_client.h

```diff
@@ -247,6 +247,22 @@ IotHttpsReturnCode_t IotHttpsClient_InitializeRequest(IotHttpsRequestHandle_t *
  * The remaining length, after the header is added, is printed to the system configured standard debug output when 
  * IOT_LOG_LEVEL_HTTPS is set to IOT_LOG_DEBUG in iot_config.h. 
  * 
+ * For an asynchronous request, this function can be invoked before the request is sent with 
+ * @ref https_client_function_sendasync, or during the @ref https_client_function_appendheadercallback. It is 
+ * recommended to invoke this function in #IotHttpsClientCallbacks_t.appendHeaderCallback.
```

Comment:
For the case where we are reusing a request that was sent synchronously. Since headers are written to the header buffer space in the user buffer provided for the request, I do not stop users from adding headers before they invoke https_client_function_sendasync. Since I do not stop users from doing that, I call out that it can be done here.